### PR TITLE
Feature/nav 150 extract python viewer package

### PR DESCRIPTION
**Pull request by munterfi**
Still a draft, there are errors from the refactoring...

### CODE REVIEW

**Comment by munterfi** [comment]
@clukas1, I think we could merge this and release the viewer with version 0.5.0?

File: public_transit_viewer/pages/isolines.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+    if show_markers:
+
+        if filter_by == "connectionRound":
+            popup = f"{data_row['targetStop']} - Round {data_row['connectionRound']}"
+        else:
+            popup = f"{data_row['targetStop']} - {data_row['durationFromSourceInMinutes']} min"
+
+        folium.Circle(  # type: ignore
+            location=source_coordinates,  # type: ignore
```
Comment (clukas1) [inline public_transit_viewer/pages/isolines.py]: needs more logic:
`location=target_coordinates if time_type == TimeType.DEPARTURE else source_coordinates`

File: public_transit_viewer/pages/isolines.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+    source_coordinates: list[float] = [data_row["sourceLat"], data_row["sourceLon"]]
+    target_coordinates: list[float] = [data_row["targetLat"], data_row["targetLon"]]
+
+    if show_markers:
+
+        if filter_by == "connectionRound":
+            popup = f"{data_row['targetStop']} - Round {data_row['connectionRound']}"
+        else:
+            popup = f"{data_row['targetStop']} - {data_row['durationFromSourceInMinutes']} min"
```
Comment (clukas1) [inline public_transit_viewer/pages/isolines.py]: same here, targetStop if timeType == DEPARTURE else sourceStop

File: public_transit_viewer/pages/isolines.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+) -> None:
+    arrival_time = data_row["durationFromSourceInMinutes"]  # type: ignore
+    color = get_color_map_hex_value(arrival_time, 0, this_filter_value[1])  # type: ignore
+
+    time_left = this_filter_value[1] - arrival_time  # type: ignore
+    radius = time_left * this_walking_speed  # type: ignore
+
+    # create a circle with the radius of the remaining distance
+    folium.Circle(  # type: ignore
+        location=[data_row["targetLat"], data_row["targetLon"]],  # type: ignore
```
Comment (clukas1) [inline public_transit_viewer/pages/isolines.py]: also time type specific:
DEPARTURE == target else source

File: public_transit_viewer/pages/isolines.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        radius=radius,  # type: ignore
+        color=color,
+        fill=True,
+        fill_color=color,
+        fill_opacity=0.3,
+        popup=f"{data_row['targetStop']} - {arrival_time} minutes",  # type: ignore
```
Comment (clukas1) [inline public_transit_viewer/pages/isolines.py]: same logic issue

**Review by clukas1 — CHANGES_REQUESTED** [review]
@munterfi nice work. No real comments on your work. Just noticed a bug that marker points were not showing correctly depending on time_type. So think this could easily be fixed in this commit before we release 0.5.0

File: public_transit_viewer/pages/isolines.py (hunk 4 above)
Comment (munterfi) [reply public_transit_viewer/pages/isolines.py]: [b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

File: public_transit_viewer/pages/isolines.py (hunk 3 above)
Comment (munterfi) [reply public_transit_viewer/pages/isolines.py]: [b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

File: public_transit_viewer/pages/isolines.py (hunk 1 above)
Comment (munterfi) [reply public_transit_viewer/pages/isolines.py]: [b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

File: public_transit_viewer/pages/isolines.py (hunk 2 above)
Comment (munterfi) [reply public_transit_viewer/pages/isolines.py]: [b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

**Review by clukas1 — APPROVED** [review]

**Pull request merged**